packages/mui-joy/src/Menu/Menu.tsx

```diff
@@ -148,6 +131,26 @@ const Menu = React.forwardRef(function Menu(inProps, ref) {
     [modifiersProp],
   );
 
+  const [SlotRoot, rootProps] = useSlot('root', {
+    ref,
+    additionalProps: {
+      anchorEl,
+      open,
+      disablePortal,
+      keepMounted,
+      as: PopperUnstyled,
```

Comment:
This refactoring of `Menu` is part of #36599. #36599 must be merged before this PR is merged.